### DISCUSSION TITLE
Can I run multiple cases simultaneously?

### DISCUSSION THREAD

**Discussion by forsyth2** [E3SM-Project/zppy#420] [Questions] [answered]
Can I run multiple cases simultaneously?

**Comment by forsyth2** [comment, marked as answer]
There are two ways of interpreting this:
1. Running multiple cases from a single `zppy` `cfg`. This not currently supported. `case` is set to be a single string in https://github.com/E3SM-Project/zppy/blob/main/zppy/templates/default.ini.
2. Running jobs from two `zppy` `cfg`s simultaneously. This is in theory possible, but be sure the `cfg`s point to different output directories. I personally don't test this often and have in the past encountered infrequent but irreproducible issues with this. It is safest to run one `cfg` at a time, at the moment.

A reasonable follow-up question then is how to run model-vs-model (as asked in #419). The best solution is to run a `zppy` `cfg` for one simulation and then use those paths as reference for the following `cfg`. Note that the integration test (the `[[ atm_monthly_180x360_aave_mvm ]]` section of https://github.com/E3SM-Project/zppy/blob/main/tests/integration/generated/test_complete_run_chrysalis.cfg) manages to not need a previous run by simply running the `test` and `ref` on different years (1852-1853 and 1850-1851 respectively) of the same simulation.

**Comment by BunnyVon** [comment]
I wonder if you have a model intercomparisons strategy on more than two cases.

**Reply by forsyth2** [reply]
@chengzhuzhang @golaz I remember seeing a demonstration of a multi-viewer for E3SM Diags (i.e., plots for multiple simulations all on the same page). However, I'm not sure where to find more info on that.

In any case, that multi-viewer isn't built into `zppy`. We can consider adding it -- I'm not sure what the level of work required would be though.

**Reply by chengzhuzhang** [reply]
Hi @BunnyVon, I'm wondering for model inter-comparison, what capabilities are you looking for. Within E3SM, there is a capability, called IICE (https://acme-climate.atlassian.net/wiki/spaces/IPD/pages/3718709487/IICE+Interface+for+Inter-Comparison+of+E3SM+diagnostics) which is developed around visualizing images from multiple E3SM Diags and Mpas-Analysis runs. On the other hand, diagnostics packages like ILAMB and PMP( PCMDI's diagnostics package) provide summary metrics plots, both packages are supported in E3SM unified enviroment, but we don't have a automated workflow to generate plots using these package, you could use zppy to generate intermediate files (those CMIP like time-series), but will need to refer to each of this package to make inter-comparison.

**Reply by BunnyVon** [reply]
Nice! I will look into it. Thank you both very much again!

**Comment by mahf708** [comment]
> Running jobs from two `zppy` `cfg`s simultaneously. This is in theory possible, but be sure the `cfg`s point to different output directories. I personally don't test this often and have in the past encountered infrequent but irreproducible issues with this. It is safest to run one `cfg` at a time, at the moment.

User feedback: I do this (*simultaneously*) all the time, and it works perfectly for me! In my use case, I am using this for the model-vs-model stuff, and `mvm` is triggered in one of the `cfg` actually depends on jobs from other `cfg`. Again, works like magic! It waits in the queue until the jobs are done and then executes the `mvm` stuff. 

In the near future, I could make my scripts available as examples... they are a total mess now because I am trying to expand the functionality of the `mvm` logic (which I love!)

**Reply by forsyth2** [reply]
That's great to hear, thanks! It's very possible the issues I ran into a while ago where completely unrelated to actually running jobs from different `cfg`s simultaneously. I usually just test separately to ensure I know what I'm testing.
